src/main/java/com/commercetools/sync/services/impl/ShoppingListServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package com.commercetools.sync.services.impl;
+
+import com.commercetools.sync.services.ShoppingListService;
+import com.commercetools.sync.shoppinglists.ShoppingListSyncOptions;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.shoppinglists.ShoppingList;
+import io.sphere.sdk.shoppinglists.ShoppingListDraft;
+import io.sphere.sdk.shoppinglists.commands.ShoppingListCreateCommand;
+import io.sphere.sdk.shoppinglists.commands.ShoppingListUpdateCommand;
+import io.sphere.sdk.shoppinglists.expansion.ShoppingListExpansionModel;
+import io.sphere.sdk.shoppinglists.queries.ShoppingListQuery;
+import io.sphere.sdk.shoppinglists.queries.ShoppingListQueryBuilder;
+import io.sphere.sdk.shoppinglists.queries.ShoppingListQueryModel;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Implementation of ShoppingListService interface.
+ */
+public final class ShoppingListServiceImpl extends BaseService<ShoppingListDraft, ShoppingList, ShoppingListSyncOptions,
+    ShoppingListQuery, ShoppingListQueryModel, ShoppingListExpansionModel<ShoppingList>>
+    implements ShoppingListService {
+
+    public ShoppingListServiceImpl(@Nonnull final ShoppingListSyncOptions syncOptions) {
+        super(syncOptions);
+    }
+
+    @Nonnull
+    @Override
+    public CompletionStage<Map<String, String>> cacheKeysToIds(@Nonnull final Set<String> shoppingListKeys) {
+
+        return cacheKeysToIds(
+            shoppingListKeys, shoppingList -> shoppingList.getKey(), keysNotCached -> ShoppingListQueryBuilder
```

Comment:
We can use method reference here to shorten the code
```suggestion
            shoppingListKeys, ShoppingList::getKey, keysNotCached -> ShoppingListQueryBuilder
```

Reply:
4e6e78f56